### PR TITLE
fix(client): wrong position of search bar components in RTL

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref: #48101

The reset icon was stacked on top of the search button in the search bar, here how it looks now

How it looks in the layout (mobile, table and desktop). I don't think I missed a layout, but please check it.

image

When the width was `768px` the search results, it would move to the left side and stack on top of the menu.

here how it looks now.

image

### CODE REVIEW

File: client/src/components/layouts/rtl-layout.css
Comment (ShaunSHamilton) [inline client/src/components/layouts/rtl-layout.css:116]: This is quite a magical number. I think we need to find a solution that does not involve hard-coding values.

Comment (Sboonny) [reply client/src/components/layouts/rtl-layout.css:116]: Yeab I have it, I am planning to open two issues after this.

But the short version is to change to grid and use pseudo class to have the input border

Comment (Sboonny) [reply client/src/components/layouts/rtl-layout.css:116]: Opened issue for this https://github.com/freeCodeCamp/freeCodeCamp/issues/48912, and if the camper is having issue with their PR, I will try to guide them instead of assuming, it won't deal well with conflicts

**Review by moT01 — COMMENTED** [review]
~~It looks a little off at a medium width window:~~
Screen Shot 2023-01-03 at 10 15 49 PM

~~And it's still stacked on top of the search icon on a small window:~~
Screen Shot 2023-01-03 at 10 16 15 PM

~~I think we could probably clean those up here.~~

And at that medium width, the search bar look pushed off to the right - it's hiding behind the scroll bar a little bit and has a larger margin on the left. Probably a separate issue. Do you want to add that to your list @Sboonny or should I make an issue for it?

**Review by moT01 — APPROVED** [review]
K, scratch those issues I mentioned. I was looking at it before this fix.

If we're fine with the hard-coded values for now, then it LGTM 🎉

**Comment by Sboonny** [comment]
> Do you want to add that to your list @Sboonny 

The issue is a bit magical to me, as it doesn't persist to the production environment.

Note: I have been trying to make sure it doesn't persist, but the searchbar's result doesn't function anymore in /learn. This does apply for the production environment, because it doesn't work there too, not the development environment only.

---

/news search bar https://www.freecodecamp.org/news

image

/learn search bar https://www.freecodecamp.org/

image

**Comment by raisedadead** [comment]
> The issue is a bit magical to me

It's not magic. Our nav and search look similar on most pages; they are not truly the same.

Have you considered the possibility that these two search bars, namely `/news` and `/learn`, are two different codebases (one is bare HTML and the other is React)?

Even the CSS (as in the resets, base templates, etc.) are slightly different and can cause differences in the behavior.

**Comment by Sboonny** [comment]
> It's not magic. Our nav and search look similar on most pages; they are not truly the same.

It's magical for me because:

- if I start my dev environment, before I have opened production environment. the search result will never show up (bug). to fix it, I need to restart my laptop or clear the browser caching, then open the production environment without running the dev environment.
- the searchbar width is different to me in dev environment than the production environment.
- the width bug which should exist in the production environment, doesn't exist in it, and I can't try to research it in there.

---

> Have you considered the possibility that these two search bars, namely /news and /learn, are two different codebases (one is bare HTML and the other is React)?

I didn't mention them because of the styling, I mentioned them because we are using the same keys for both, I think 🤔, but still the searchbar in `/learn` is buggy and magical to me.

I mentioned it, because I thought we hit rate-limit in `/learn` or something.

**Comment by Sboonny** [comment]
> Even the CSS (as in the resets, base templates, etc.) are slightly different and can cause differences in the behavior.

I was wondering, why you are mentioning CSS files, while I am talking about search result. looking back at the picture it doesn't include search result in it 😅 🤣.

Give me a minute, let me increase the width 👍.

**Comment by Sboonny** [comment]
Fixed:

- menu aligning to the right when the screen width bigger than `600` but lower than `767`
- fix the language icon and night mode icon rotating in the menu

image

---

I didn't find the bug from width `980` to width `600` in edge and firefox, is the bug still there?

image

**Review by raisedadead — APPROVED** [review]

**Review by moT01 — CHANGES_REQUESTED** [review]
Something happened here with the last commit I think that made the search results appear not below the input at medium sized window widths...
Screen Shot 2023-01-05 at 11 12 13 PM

**Comment by Sboonny** [comment]
and the width still overflowing to the right. 😞

**Review by moT01 — APPROVED** [review]
LGTM 🎉

The scrollbar issue is pretty minor and I think it was pre-existing - so I don't think it's necessary to fix it here. I would maybe save it for another PR since this one is ready to merge.

**Comment by Sboonny** [comment]
I have added a quick hack to account for scroll bar spacing, but this should be reverted in future PR, and remove some of the divs element, so the flex can actually flex and follow the parent width, this will make it follow the parent width and respect the scroll bar (when it's there) at the same time.

**Review by scissorsneedfoodtoo — COMMENTED** [review]
Tested this locally and everything LGTM 👍 👍 

Played around with different widths and the RTL search bar result and reset icons seem to mirror those in the LTR production search bar.

Looking at https://github.com/freeCodeCamp/freeCodeCamp/pull/48944/files, we might want to wait for that to land before merging this. I suspect there'll be some merge conflicts in that PR if we merge this first.

Also, we don't have any RTL languages live on Learn yet, so it might be better to get https://github.com/freeCodeCamp/freeCodeCamp/pull/48944/files in first, then fix any conflicts in this PR.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Everything LGTM 👍 👍 

Played around with the simple search bar on / and the search bar with the dropdown on /learn, and the search bar, reset button, and dropdown all look good at different screen sizes.

I've just got one minor suggestion I'll leave as a review comment.

File: client/src/components/layouts/rtl-layout.css
Comment (scissorsneedfoodtoo) [inline client/src/components/layouts/rtl-layout.css:103]: It seems like we want to apply these styles on top of the one in this media query: https://github.com/freeCodeCamp/freeCodeCamp/blob/fdbc74999f4147b6446d75b1c53d10976cfab390/client/src/components/Header/components/universal-nav.css#L227

It might make sense to target the same `min-width` here so it's clear there's a connection between the two media queries.

```suggestion
@media (min-width: 601px) {
```

Comment (scissorsneedfoodtoo) [inline client/src/components/layouts/rtl-layout.css:103]: It seems like we want to apply these styles on top of the styles in this media query in `universal-nav.css`: https://github.com/freeCodeCamp/freeCodeCamp/blob/fdbc74999f4147b6446d75b1c53d10976cfab390/client/src/components/Header/components/universal-nav.css#L227

If that's true, it might make sense to target the same `min-width` so that it's clear both will be applied under the same condition.

```suggestion
@media (min-width: 601px) {
```

**Review by ojeytonwilliams — APPROVED** [review]

**Pull request merged**
